### PR TITLE
[TrilinosApplication] Adding `SetValue` methods and corresponding tests for `TrilinosAssemblingUtilities`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This commit introduces new functionality to the `TrilinosAssemblingUtilities` class by adding `SetValue` methods for both vectors and matrices. The methods include options for global and local indexing, as well as global and local assembling.

The changes involve the `trilinos_assembling_utilities.h file, which define the new `SetValue` methods. Furthermore, the PR includes a new test file, `test_trilinos_assembling_utilities.cpp`, with distributed test cases to check the correct functionality of the newly added methods.

In summary, this commit adds new `SetValue` methods to the `TrilinosAssemblingUtilities` class and corresponding tests to verify their functionality.

**🆕 Changelog**

- [Adding `SetValue` methods to `TrilinosAssemblingUtilities`](https://github.com/KratosMultiphysics/Kratos/commit/2142bd50b262f86d10724d648d1afc81b0909725)
- [Adding tests for `TrilinosAssemblingUtilities`](https://github.com/KratosMultiphysics/Kratos/commit/a1133cc039c108e55a4ae1cf7aaa9d222600db4b)

### CODE REVIEW

**Comment by loumalouomega** [comment]
Done @jcotela

**Review by jcotela — APPROVED** [review]
👍 looks good, thanks for the patience

**Comment by loumalouomega** [comment]
> 👍 looks good, thanks for the patience
